src/org/javaday/java8/misused/stream/DoNotNeglectDataStructures.java

```diff
@@ -0,0 +1,44 @@
+package org.javaday.java8.misused.stream;
+
+import org.javaday.java8.misused.Annotations.Good;
+import org.javaday.java8.misused.Annotations.Ugly;
+
+import java.util.List;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toList;
+
+//Taken from real code
+public class DoNotNeglectDataStructures {
+
+    @Ugly
+    class UnnecessaryUseOfNestedStreamOperations {
+
+        public List<Order> filterOrdersByStatuses(List<Order> orders, Set<Status> appropriateStatuses) {
```

Comment:
I would leave it as is as it's taken from real code. With `User#Set<Role>` it will be a bit different code. Feel free to play with it if you like though.